analyzers/src/SonarAnalyzer.Common/Helpers/KnownReference.Predicates.cs

```diff
@@ -0,0 +1,74 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2023 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+namespace SonarAnalyzer.Helpers;
+
+public sealed partial class KnownReference
+{
+    internal static Func<AssemblyIdentity, bool> NameIs(string name) =>
+        x => x.Name.Equals(name);
```

Comment:
I am not sure about the casing of assembly names. Currently, I use the standard string.Equals but this is most likely wrong:

> The runtime treats assembly names as case-insensitive when binding to an assembly, but preserves whatever case is used in an assembly name. Several tools in the Windows SDK handle assembly names as case-sensitive. For best results, manage assembly names as though they were case-sensitive.

[Source](https://learn.microsoft.com/en-us/dotnet/standard/assembly/names)

As far as I know, you can only load assemblies in the run-time with unique names, and the casing is ignored. I think it would be better to do `StringComparison.OrdinalIgnoreCase` but I'm not sure.

Reply:
Let's play safe and use case insensitive. I can't imagine that xUnit 1.0 would be something else than xunit 1.0